test/app/features/splash/splash_page_test.dart

```diff
@@ -0,0 +1,232 @@
+              .thenAnswer((i) => Future.value());
+
+          await theAppIsRunning(tester, const SplashPage());
+          verify(() => AppModulesMock.modularNavigator
+              .pushReplacementNamed('/authentication')).called(1);
+        },
+      );
+
+      group('in authenticated states', () {
+        testWidgets(
+          'with ServerSideSessionFailed forward to `authentication`',
+          (tester) async {
+            when(() => AppModulesMock.appConfiguration.authorizationStatus)
+                .thenAnswer((i) async => AuthorizationStatus.authenticated);
+            when(() => AppModulesMock.appStateUseCase.check())
+                .thenFailure((_) => ServerSideSessionFailed());
+            when(() =>
+                    AppModulesMock.modularNavigator.pushReplacementNamed(any()))
+                .thenAnswer((i) => Future.value());
+
+            await theAppIsRunning(tester, const SplashPage());
+            verify(() => AppModulesMock.modularNavigator
+                .pushReplacementNamed('/authentication')).called(1);
+          },
+        );
+        testWidgets(
+          'for unmapped error and in stealth mode forward to `stealthModeEnabled`',
+          (tester) async {
+            when(() => AppModulesMock.appConfiguration.authorizationStatus)
+                .thenAnswer((i) async => AuthorizationStatus.authenticated);
+
+            when(() => AppModulesMock.appStateUseCase.check())
+                .thenFailure((_) => ServerFailure());
+
+            when(() => AppModulesMock.userProfileStore.retrieve()).thenAnswer(
+              (_) async => FakeUserProfileEntity.stealth(),
+            );
+
+            when(() =>
+                    AppModulesMock.modularNavigator.pushReplacementNamed(any()))
+                .thenAnswer((i) => Future.value());
+
+            await theAppIsRunning(tester, const SplashPage());
+            verify(() => AppModulesMock.modularNavigator
+                .pushReplacementNamed('/authentication/stealth')).called(1);
+          },
+        );
+
+        testWidgets(
+          'for unmapped error and in anonymous mode forward to `anonymousModeEnabled`',
+          (tester) async {
+            when(() => AppModulesMock.appConfiguration.authorizationStatus)
+                .thenAnswer((i) async => AuthorizationStatus.authenticated);
+
+            when(() => AppModulesMock.appStateUseCase.check())
+                .thenFailure((_) => ServerFailure());
+
+            when(() => AppModulesMock.userProfileStore.retrieve()).thenAnswer(
+              (_) async => FakeUserProfileEntity.anonymous(),
+            );
+
+            when(() =>
+                    AppModulesMock.modularNavigator.pushReplacementNamed(any()))
+                .thenAnswer((i) => Future.value());
+
+            await theAppIsRunning(tester, const SplashPage());
+            verify(() => AppModulesMock.modularNavigator
+                    .pushReplacementNamed('/authentication/sign_in_stealth'))
```

Comment:
Para o teste 'for unmapped error and in anonymous mode forward to `anonymousModeEnabled`', pq verifica se o pushReplacementNamed foi no '/authentication/sign_in_stealth'? Isso não seria apenas para o `stealthModeEnabled`'?

Reply:
Este fluxo é definido pelo `_userProfileStore.retrieve()`, e neste momento estou recebendo do mock que o perfil está sinalizando como `anonymousModeEnabled` que direciona para o `/authentication/sign_in_stealth`.  Quando está no `estado` `stealthModeEnabled` o fluxo é redirecionado para o `/authentication/stealth`